### ISSUE TITLE
Use NeoForge server.jar to allow using our own LaunchServer scripts

### ISSUE THREAD

**Issue by RyanTheAllmighty** [ATLauncher/ATLauncher#921]
### The Problem

Running servers for NeoForge and Forge that use the run scripts are a pain, we currently do not do our own special sauce like we do for other packs which leads to many support requests regarding these servers not running and trying to guide users to edit run.bat and install proper Java versions

### The Solution

NeoForge have released https://github.com/neoforged/ServerStarterJar which aims to be a single jar that can be run.

The launcher config has this option under `neoForgeServerStarterJar` key with below object:

```json
{
    "enabled": false,
    "url": "https:\/\/github.com\/neoforged\/ServerStarterJar\/releases\/download\/0.1.19\/server.jar",
    "hash": "49f6b85178ea20ee79e199b6f8cce884b1f510235bcb41add1f6ce04d2539238",
    "size": 11912
}
```

Using this (if enabled) when there are run.bat/sh scripts created, simply download the server.jar into the server folder, and then we can use our own LaunchServer scripts which has some nice things in it to auto use the correct java and launch from the launcher correctly.

Just needs to be run like `java @user_jvm_args.txt -jar server.jar` so the jvm args file is correctly loaded

The original run scripts need to be in the folder still, so the folder will have the forge scripts, as well as the LaunchServer scripts, but the idea is that either way no matter which script is used, it will work, and having the LaunchServer scripts allows it to just work correctly with the launchers server manager

### Have you checked this issue doesn't already exist?

- [X] I am sure this issue doesn't exist already

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RyanTheAllmighty** [comment]
> Heyo,
> 
> chiming in to warn you about some issues you may run into when using the ServerStarterJar. I'm making use of it in ServerPackCreator, too, and had to deal with the following:
> 
> 1. Use a link to the installer for Forge
> 2. Use the modloader version for NeoForge MC 1.20.2 and up
> 3. Some NeoForge versions require a link to the installer still, mostly for MC 1.20.1 and down
> 4. When using the SSJ for Forge, keep in mind that the Forge installers terminate the JVM after installation. You can avoid this via use of `-Djava.security.manager=allow` to allow the SSJ to catch the `System.exit()` of the installer.
> 
> If you want some examples, you may take a look at my scripts, especially the `SetupForge` and `SetupNeoForge` functions:
> 
> * Shell/Bash: https://github.com/Griefed/ServerPackCreator/blob/main/serverpackcreator-api/src/main/resources/de/griefed/resources/server_files/default_template.sh
> * PowerShell: https://github.com/Griefed/ServerPackCreator/blob/main/serverpackcreator-api/src/main/resources/de/griefed/resources/server_files/default_template.ps1
> 
> Cheers, Griefed

Thanks for that. We already do all the installer running on the launcher side before creating the server files, so we don't have that headache, so far it seems simple enough to just be able download the server.jar and then use it as the main jar to run and it seems to work if you already have all the libraries where they are expected (which we already do anyway).

Our server start scripts are very simple, just to allow us to pass a Java path to use then forward on so that we can avoid having people manage their own Java and instead use the Mojang provided ones that we already download through the launcher.

**Comment by Griefed** [comment]
Glad to hear that. :)

If you ever decide to upgrade to Java 21, give me a ping. Would be fun to see how I could potentially integrate the ServerPackCreator's API into ATLauncher, so people can hit a button "Generate Server Pack" for their modpacks. :D

**Comment by RyanTheAllmighty** [comment]
Yeah I took a look at your repo and expected it was just a docker init script but looks so much more.

Definitely keep my eye on it as server generation is not always great, mainly looking at CurseForge for fumbling that up
